csrc/quantization/cutlass_w8a8/scaled_mm_c2x.cu

```diff
@@ -252,14 +313,13 @@ void cutlass_gemm_caller(torch::Tensor& out, torch::Tensor const& a,
 
 }  // namespace
 
-void cutlass_scaled_mm_sm75(torch::Tensor& out, torch::Tensor const& a,
-                            torch::Tensor const& b,
-                            torch::Tensor const& a_scales,
-                            torch::Tensor const& b_scales) {
+template <template <typename, typename> typename Epilogue,
+          typename... EpilogueArgs>
+void cutlass_scaled_mm_sm75_epilogue(torch::Tensor& out, torch::Tensor const& a,
+                                     torch::Tensor const& b,
+                                     EpilogueArgs&&... epilogue_args) {
   TORCH_CHECK(a.dtype() == torch::kInt8);
   TORCH_CHECK(b.dtype() == torch::kInt8);
-  TORCH_CHECK(a_scales.dtype() == torch::kFloat32);
-  TORCH_CHECK(b_scales.dtype() == torch::kFloat32);
```

Comment:
Is there a reason to remove the dtype checking of a_scales & b_scales? There are some that check this and some don't. Should this be moved to the entry function?

Reply:
Yeah, this became a templated function which does not know about `a_scales` and `b_scales` anymore. The new entry point now does the check ([below](https://github.com/vllm-project/vllm/pull/5560/files/97158de203a7769e29967c67cdf2ff190d4f695a#diff-adbdc77eb328148f4530aaeea7d938138824bff858a86cbbff2454443011dde2R347-R348))